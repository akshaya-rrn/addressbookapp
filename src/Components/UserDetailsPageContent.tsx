import { Alert, AlertTitle, Box, Typography } from '@mui/material';
import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import { UserResponseResult } from '../types';
import { formatDOB, getUserName } from '../utility/Utils';

interface UserDetailedContentType {
    user?: UserResponseResult | null;
}

export const UserDetailsPageContent: React.FC = () => {

    const location = useLocation();
    const stateData = location.state as UserDetailedContentType ;
    if (stateData == null) {
        return <AlertComponent/>
    }
    const { name, cell, dob } = stateData.user as UserResponseResult;
    
    return <Link to='/detailedview'>
    <Box sx={{ maxWidth: 400}} >
        <Typography id="name" variant="h6" component="div">Name : {getUserName(name)}</Typography>
        <Typography id="cell-number" sx={{ mt: 2 }}>Cell :{cell}</Typography>
        <Typography id="DOB" sx={{ mt: 2 }}>DOB :{formatDOB(dob.date)}</Typography>
    </Box>
    </Link>; 
}

const AlertComponent: React.FC = () => {
    return <Alert severity="warning">
    <AlertTitle>Invalid Request</AlertTitle>
    There is something wrong with the request â€” <strong>Please try again!</strong>
  </Alert>
}