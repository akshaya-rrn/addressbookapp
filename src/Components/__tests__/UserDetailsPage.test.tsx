import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';


import { UserDetailsPageContent } from '../UserDetailsPageContent';
import { mockUserData } from './mockData/getUsersData';


describe('Users Details Component when state data is undefined', () => {
  const mockLocationData = { pathname: '', state: null }

  beforeEach(() => {
    //useLocation.mockReturnValue(mockLocationData as any);
    jest.mock("react-router-dom", () => ({
      ...jest.requireActual("react-router-dom"),
      useLocation: () => (mockLocationData)
    }));
  });

  it('Users Card component is rendered correctly when value passed as null', () => {
    const tree = renderer.create(<Router><UserDetailsPageContent /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Users Details Component when userData is passed', () => {
  const mockLocationData = { pathname: 'localhost:3000/detailedview', state: mockUserData }

  beforeEach(() => {
  jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => (mockLocationData)
    }));
  });

  it('Users Card component is rendered correctly when value passed', () => {
    const tree = renderer.create(<Router><UserDetailsPageContent /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

