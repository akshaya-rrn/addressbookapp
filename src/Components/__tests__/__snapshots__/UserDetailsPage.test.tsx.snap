// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Details Component when state data is undefined Users Card component is rendered correctly when value passed as null 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-rob5aa-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ReportProblemOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      Invalid Request
    </div>
    There is something wrong with the request — 
    <strong>
      Please try again!
    </strong>
  </div>
</div>
`;

exports[`Users Details Component when userData is passed Users Card component is rendered correctly when value passed as null 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-rob5aa-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ReportProblemOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      Invalid Request
    </div>
    There is something wrong with the request — 
    <strong>
      Please try again!
    </strong>
  </div>
</div>
`;
